test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -753,11 +772,121 @@ public ReferenceExpression visit(ReferenceExpression r)
         return process(Who.cas, updatedCondition, lets);
     }
 
+    public Consumer<ThrowableAssert.ThrowingCallable> shouldReject(Txn txn)
+    {
+        if (!shouldApply(txn)) return null;
+        List<Mutation> mutations = null;
+        if (txn.ifBlock.isPresent()) mutations = txn.ifBlock.get().mutations;
+        else if (!txn.mutations.isEmpty()) mutations = txn.mutations;
+
+        if (mutations == null) return null;
+        Set<Consumer<Throwable>> checks = new HashSet<>();
+        for (var m : mutations)
+        {
+            var failures = shouldReject0(m);
+            if (failures != null)
+                checks.addAll(failures);
+        }
+        return toShouldRejectConsumer(checks);
+    }
+
+    public Consumer<ThrowableAssert.ThrowingCallable> shouldReject(Mutation mutation)
+    {
+        return toShouldRejectConsumer(shouldReject0(mutation));
+    }
+
+    private static Consumer<ThrowableAssert.ThrowingCallable> toShouldRejectConsumer(@Nullable Set<Consumer<Throwable>> checks)
+    {
+        return checks == null || checks.isEmpty() ? null
+                                                  : t -> Assertions.assertThatThrownBy(t)
+                                                                   .satisfiesAnyOf(checks.toArray(Consumer[]::new));
+    }
+
+    private Set<Consumer<Throwable>> shouldReject0(Mutation mutation)
+    {
+        if (!shouldApply(mutation)) return null;
+        Set<Consumer<Throwable>> checks = null;
+        if (mutation.kind == Mutation.Kind.UPDATE)
```

Comment:
these checks are way too specific... need to figure out a better way to make this easier to maintain